### PR TITLE
added usage example to pad_to_bounding_box()

### PR DESCRIPTION
**Pull request by generationzcode**
I added a usage example to tf.image.pad_to_bounding_box() function. I tested it on colab so I know there aren't syntax errors.😄🖖

### CODE REVIEW

**Comment by generationXcode** [comment]
I signed it!

**Comment by kyscg** [comment]
@googlebot I signed it!

File: tensorflow/python/ops/image_ops_impl.py [hunk 1]
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
       `target_*` arguments, or either `offset_height` or `offset_width` is
       negative.
+      
+  Usage example:
+    import tensorflow as tf
```
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    
```

File: tensorflow/python/ops/image_ops_impl.py [hunk 2]
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
       `target_*` arguments, or either `offset_height` or `offset_width` is
       negative.
+      
+  Usage example:
```
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
  Usage example:
  ```python
```

File: tensorflow/python/ops/image_ops_impl.py [hunk 3]
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
...
+    import tensorflow as tf
+    x=tf.random.uniform((28,28,1))
```
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> x = tf.random.uniform((28, 28, 1))
```

File: tensorflow/python/ops/image_ops_impl.py [hunk 4]
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
...
+    tf.image.pad_to_bounding_box(x,28,28,100,100)
```
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100)
```

File: tensorflow/python/ops/image_ops_impl.py [hunk 5]
```diff
@@ -792,6 +792,12 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
...
+
```
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    ```
    
```

**Review by kyscg — CHANGES_REQUESTED** [review]
There is no need to import tensorflow everywhere, please make these changes

**Comment by generationXcode** [comment]
Done it!

File: tensorflow/python/ops/image_ops_impl.py [hunk 6]
```diff
@@ -792,6 +792,14 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
       `target_*` arguments, or either `offset_height` or `offset_width` is
       negative.
+      
+  Usage example:
+  ```python
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100)
+    ```
```
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Assert something about the result to make this a doctest? Like print its shape at least?

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: Ok!

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: done!😊

Comment (alextp) [reply tensorflow/python/ops/image_ops_impl.py]: Then put the resulting shape in the comment so this runs as a doctest

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: done! If there are any other changes needed please say so. I can't wait to see this merged😊

File: tensorflow/python/ops/image_ops_impl.py [hunk 7]
```diff
@@ -792,6 +792,15 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
       `target_*` arguments, or either `offset_height` or `offset_width` is
       negative.
+      
+  Usage example:
+  ```python
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    >>> # TensorShape([100, 100, 1])
```
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Why is the result commented? You can remove the `>>> #` and this will make the doctest work.

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: Did it!

**Review by alextp — DISMISSED** [review]

File: tensorflow/python/ops/image_ops_impl.py [hunk 8]
```diff
@@ -773,7 +773,7 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
 
   This op does nothing if `offset_*` is zero and the image already has size
   `target_height` by `target_width`.
-
+
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Please remove all trailing whitespace

File: tensorflow/python/ops/image_ops_impl.py [hunk 9]
```diff
@@ -792,6 +792,15 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
       `target_*` arguments, or either `offset_height` or `offset_width` is
       negative.
+      
+  Usage example:
+  ```python
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: No need to add the ` ```python` and ` ``` ` markers. Just have an empty line after `Usage example:` and after the code block. Move the doctest above, just before the `Args:` section

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/image_ops_impl.py (hunk 8 above)
Comment (mihaimaruseac) [reply tensorflow/python/ops/image_ops_impl.py]: The whitespace is not removed
Screenshot from 2020-01-04 13-14-23

File: tensorflow/python/ops/image_ops_impl.py [hunk 10]
```diff
@@ -790,8 +790,17 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
...
   Raises:
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
-      `target_*` arguments, or either `offset_height` or `offset_width` is
-      negative.
+    `target_*` arguments, or either `offset_height` or `offset_width` is
+    negative.
+      
+  Usage example:
+  
+  python
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Why is this `python` here?

File: tensorflow/python/ops/image_ops_impl.py [hunk 11]
```diff
@@ -790,8 +790,17 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
...
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    >>> TensorShape([100, 100, 1])
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Why are these here?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please consult https://www.tensorflow.org/community/contribute/docs_ref and follow the guideliness from there

File: tensorflow/python/ops/image_ops_impl.py (hunk 11 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: Its the usage example

File: tensorflow/python/ops/image_ops_impl.py (hunk 8 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: oh sorry.

**Comment by generationXcode** [comment]
Done!

File: tensorflow/python/ops/image_ops_impl.py (hunk 11 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: But I think you might be referring to the `TensorShape([100, 100, 1])`... Umm in this commit I removed the `>>>` that comes before it, so everyone knows its the output.

File: tensorflow/python/ops/image_ops_impl.py [hunk 12]
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
 def pad_to_bounding_box(image, offset_height, offset_width, target_height,
                         target_width):
   """Pad `image` with zeros to the specified `height` and `width`.
-
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Why did you remove this line?

File: tensorflow/python/ops/image_ops_impl.py [hunk 13]
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
...
   Adds `offset_height` rows of zeros on top, `offset_width` columns of
   zeros on the left, and then pads the image on the bottom and right
   with zeros until it has dimensions `target_height`, `target_width`.
-
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Why did you remove this line?

File: tensorflow/python/ops/image_ops_impl.py [hunk 14]
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
...
   This op does nothing if `offset_*` is zero and the image already has size
   `target_height` by `target_width`.
-
+  Usage example:
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Empy line above.

File: tensorflow/python/ops/image_ops_impl.py [hunk 15]
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
...
+    
+    >>> x = tf.random.uniform((28, 28, 1))
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100).shape
+    TensorShape([100, 100, 1])
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Empty line below

File: tensorflow/python/ops/image_ops_impl.py [hunk 16]
```diff
@@ -766,32 +766,32 @@ def _get_dim(tensor, idx):
...
     `[batch, target_height, target_width, channels]`
     If `image` was 3-D, a 3-D float Tensor of shape
     `[target_height, target_width, channels]`
-
   Raises:
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
-      `target_*` arguments, or either `offset_height` or `offset_width` is
-      negative.
+    `target_*` arguments, or either `offset_height` or `offset_width` is
+    negative.
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Don't add trailing whitespace.

File: tensorflow/python/ops/image_ops_impl.py (hunk 15 above)
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: I would rather have `x` be a small constant with some nice values and then pad to a slightly larger box (instead of going from 28 to 100) and displaying the values instead of just the shape

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: you mean `tf.constant`?

File: tensorflow/python/ops/image_ops_impl.py (hunk 12 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: I fixed it.😊

File: tensorflow/python/ops/image_ops_impl.py (hunk 13 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: I fixed it.😊

File: tensorflow/python/ops/image_ops_impl.py (hunk 14 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: I fixed it.😊

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: I fixed it.😊

File: tensorflow/python/ops/image_ops_impl.py [hunk 17]
```diff
@@ -766,32 +766,36 @@ def _get_dim(tensor, idx):
 def pad_to_bounding_box(image, offset_height, offset_width, target_height,
                         target_width):
   """Pad `image` with zeros to the specified `height` and `width`.
-
+
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: dTr3qVQtZEr

File: tensorflow/python/ops/image_ops_impl.py (hunk 15 above)
Comment (mihaimaruseac) [reply tensorflow/python/ops/image_ops_impl.py]: Yes

**Review by mihaimaruseac — COMMENTED** [review]
Don't forget to remove trailing whitespace

File: tensorflow/python/ops/image_ops_impl.py (hunk 15 above)
Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: Hi! I have no problem with the changing of the data, I just dont know how to display the values of a tensor. nvm I'll find out😊

Comment (mihaimaruseac) [reply tensorflow/python/ops/image_ops_impl.py]: Just run the test on your local machine. https://www.tensorflow.org/community/contribute/docs_ref#test_on_your_local_machine

Comment (mihaimaruseac) [reply tensorflow/python/ops/image_ops_impl.py]: This is still not done

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please change doctest from testing only the shape of the tensor to also testing the values, as told in comments

**Comment by generationXcode** [comment]
@mihaimaruseac Do we use `tensor.eval()` to show the values? I tried it once, it gave me errors...

**Comment by mihaimaruseac** [comment]
No need to `tensor.eval()` (though it would have been useful to also see what errors you were getting).

You can just do something similar to

```python
>>> x = tf.constant([2,3,4])
>>> y = tf.constant([4,5,6])
>>> tf.add(x, y)
<Tensor... [6,8,10]...>
```

(where I ellided a lot of details in the output, but you get the general idea)

**Comment by generationXcode** [comment]
got it!

File: tensorflow/python/ops/image_ops_impl.py [hunk 18]
```diff
@@ -774,9 +774,19 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
...
   `target_height` by `target_width`.
 
+  Usage example:
+
+    >>> x=tf.constant([
+    ...        [[1,2,3],[4,5,6],[7,8,9]],
+    ...        [[1,2,3],[4,5,6],[7,8,9]],
+    ...        [[1,2,3],[4,5,6],[7,8,9]]
+    ...        ])
+    >>> tf.image.pad_to_bounding_box(x, 28, 28, 100, 100)
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Change the bounding box to smaller values so that we could print all the values in `numpy=array([[[...`?

Comment (generationXcode) [reply tensorflow/python/ops/image_ops_impl.py]: got it. I'll reduce the size of both the arrays.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@generationXcode thank you, it is still failing doctest can you please check here for [logs](https://source.cloud.google.com/results/invocations/383d702a-33c9-42f2-94a3-0038329704f1/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests).

Please run the doctest locally as mentioned here in the [contributor guidelines](https://www.tensorflow.org/community/contribute/docs_ref).

Also, can you please help to fix Ubuntu Sanity errors? Thanks!

**Comment by generationXcode** [comment]
Hi, sorry for such a late reply (its my computer time now). Umm I don't know much about these checks but I am doing what you said and running the doctest locally, presently I'm downloading the tensorflow repo

**Comment by gbaned** [comment]
@generationXcode  it is still failing doctest can you please check here for [logs](https://source.cloud.google.com/results/invocations/4de04c11-2db2-48f6-8ced-b4847ba81db5/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests). Thanks!

**Comment by generationXcode** [comment]
@gbaned is there any way we can run the doctest again? Thanks!

**Comment by mihaimaruseac** [comment]
Unfortunately we can only trigger all tests at once. Retriggered them now

**Comment by generationXcode** [comment]
Thanks!

**Comment by generationXcode** [comment]
it seems like the Ubuntu CPU test is contradicting the other results https://source.cloud.google.com/results/invocations/a46ee8e1-2790-41ad-8f78-216803471602/targets/%2F%2Ftensorflow%2Ftools%2Fdocs:tf_doctest/tests
If I solve this problem then the other tests will fail...

**Comment by generationzcode** [comment]
Hi, I know this pull request isn't closed, but GCI is and I'd like to thank all of the mentors for helping me during the contest and also the reviewers of the pull requests I made.😊
